packages/adblocker/src/engine/engine.ts

```diff
@@ -1214,12 +1234,24 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
       }
     }
 
-    if (allowGenericHides === true) {
-      allowGenericHides = shouldApplyHideException(genericHides) === false;
+    if (genericHideException === undefined) {
+      genericHideException = shouldApplyHideException(genericHides);
     }
 
-    if (allowSpecificHides === true) {
-      allowSpecificHides = shouldApplyHideException(specificHides) === false;
+    if (specificHideException === undefined) {
+      specificHideException = shouldApplyHideException(specificHides);
+    }
+
+    if (genericHideException !== undefined) {
+      matches.push({
+        filter: undefined,
+        exception: genericHideException,
+      });
+    } else if (specificHideException !== undefined) {
```

Comment:
cannot we have both? 

we can push the match directly after we get return value from  shouldApplyHideException.

eg.
```js
if (genericHideException === undefined) {
  genericHideException = shouldApplyHideException(genericHides);
  if (genericHideException !== undefined) {
    matches.push({ filter: undefined, exception: genericHideException });
  }
}

```

Reply:
It has to be a separate if statement as its purpose is to report the existence of `genericHideException`.